### ISSUE TITLE
Need better error message when mkdir fails

### ISSUE THREAD

**Issue by ksaur** [Azure/azure-storage-azcopy#1882]
##### Note: The version is visible when running AzCopy without any argument
azcopy_linux_amd64_10.16.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Linux

##### Note: Please remove the SAS to avoid exposing your credentials. If you cannot remember the exact command, please retrieve it from the beginning of the log file.

I tried to run both `azcopy cp` and `azcopy env`

### What problem was encountered?

```
 ./azcopy cp foo "https://myblobstore"
panic: mkdir /.azcopy: permission denied

goroutine 1 [running]:
github.com/Azure/azure-storage-azcopy/v10/common.PanicIfErr(...)
        /home/vsts/work/1/s/common/lifecyleMgr.go:685
main.main()
        /home/vsts/work/1/s/main.go:57 +0x53c
```

From reading some issues/posts, I see that I just need to set `AZCOPY_LOG_LOCATION` and `AZCOPY_PLAN_FILE_LOCATION`.

But before I set them, I tried this and it also failed:
```
./azcopy env
panic: mkdir /.azcopy: permission denied

goroutine 1 [running]:
github.com/Azure/azure-storage-azcopy/v10/common.PanicIfErr(...)
        /home/vsts/work/1/s/common/lifecyleMgr.go:685
main.main()
        /home/vsts/work/1/s/main.go:57 +0x53c
```

### Have you found a mitigation/solution?

In #903, I learned that what I needed to do was set `AZCOPY_LOG_LOCATION` and `AZCOPY_PLAN_FILE_LOCATION`.

The fact that I couldn't even call `azcopy env` without setting this was a bit confusing.

A helpful log message maybe [here](https://github.com/Azure/azure-storage-azcopy/blob/13fecd4efee985bc1f2cf03fd130fe613ecec375/main.go#L63) could fix this issue.

**Comment by adreed-msft** [comment]
I've reviewed your PR. Thank you for your contribution.

**Comment by gapra-msft** [comment]
Hi @ksaur it looks like your PR was merged and released as part of 10.16.2, so I will go ahead and close this issue. Feel free to reopen the issue in case you need further improvements with this error message. Thanks again for your contribution!

**Comment by heiner** [comment]
Just a quick note to say that the name of the second variable is actually `AZCOPY_JOB_PLAN_LOCATION` and this error message introduced here, while meant to be helpful, is somewhat misleading.
